initial commit

GlizzyProductions opened the pull request:
starting the assignment

GlizzyProductions committed:
initial commit

starting the assignment

GlizzyProductions committed:
completed 5 sub-object, including member variables and function templates. also removed compile: -s from the .replit file

GlizzyProductions committed:
completed 1b, listed 4 unrelated User Defined Types with 5 class member vairables and 3 member functions.

GlizzyProductions committed:
slight change in the student class (watch televsion)

GlizzyProductions committed:
completed 1c, defined a UDT of girlfriend with 5 sub-objects and listed data types for the sub-objects listed in 1b

GlizzyProductions committed:
completed 1d. created 5 sub-objects from the member variables from 1c

GlizzyProductions committed:
as per requirement for step 6 I relocated the 10 sub-objects  down to  line 427 and placed them in order.

matkatmusic reviewed (changes requested):
Complete the rest of the assignment

GlizzyProductions committed:
problem 1 solved... no known errors

GlizzyProductions committed:
setting up the 10 UDTs

GlizzyProductions committed:
filled out information on massage chair UDT

GlizzyProductions committed:
completed CollegeStudent sub-object main.cpp line473

GlizzyProductions committed:
Completed sub-Object PetCat main.cpp line495

GlizzyProductions committed:
completed questions

I filled out 10 user defined types complete with 5 member variables and 3 functions

GlizzyProductions committed:
completed Part 1e - step 7

GlizzyProductions committed:
created a nested UDT sub-Object (HealthStatus) for Human

GlizzyProductions committed:
created another nested UDT sub-Object (CatFoodRobot) for PetCat

GlizzyProductions committed:
completed Part 1e- Step 12 included parameter list using UDTs

GlizzyProductions committed:
finished 1e - step 15

GlizzyProductions committed:
completed 1e - step 19: pasted 10 UDTs directly above the int main( ) function.

submit a review

matkatmusic commented on main.cpp:
```diff
@@ -1010,14 +884,245 @@ Part 1e - Step 19: Request a review
+    void knockOverObjects();
+    //3) scratch visitors
+    void scratchVisitors();
+    CatFoodRobot currentCatFeeder;
```
Cats wouldn't own a `CatFoodRobot`.
a `CatOwner` would. 
remove this member variable
- [x] revise

GlizzyProductions replied:
I think I understand! So I figured cats have collars so I changed the nested class to CatCollar to make things more realistic.

matkatmusic commented on main.cpp:
```diff
@@ -1010,14 +884,245 @@ Part 1e - Step 19: Request a review
+    void goToSleep(int howLong);
+    //3) donate blood
+    void donateBlood(bool giveLeftArm);
+    HealthStatus annualCheckup;
```
an `AnnualCheckup` would provide a `HealthStatus`.  
but an `AnnualCheckup` is NOT a `HealthStatus`.  
this variable should be called `healthStatus`;
If you wanted this idea to be expressed correctly, you'd have a function like this:
```
HealthStatus goToAnnualPhysical(CollegeStudent resident);
```
where `resident` is obviously a student doctor who hasn't completed their residency yet at a hospital or clinic. 
Code represents real life concepts.  it is important that we choose names and create objects that represent real life and real life concepts.
- [x] revise

GlizzyProductions replied:
point taken, I will keep this in mind moving forward.

matkatmusic commented on main.cpp:
```diff
@@ -1010,14 +884,245 @@ Part 1e - Step 19: Request a review
+    void loseLimbs();
+};
+
+struct EducationLevel
```
`EducationLevel` is a single thing.  `Education` encompasses many things.
This class should be called `Education`
- [x] revise

GlizzyProductions committed:
altered UDT PetCat and the nested UDT to represent a CatCollar for the purpose of being more realistic

GlizzyProductions committed:
made requested revisions from matkat

changed 'EducationLevel' to 'Education' also replaced 'annualCheckUp' member variable with 'healthStatus' as suggested by matkat.
